Develop

NataliaN951 opened the pull request:
Обязательное задание
1. Привяжи GitHub к тренажёру. Как только ты это сделаешь, в списке репозиториев автоматически появится qa_java. Там будет заготовка проекта, в которой нужно дописать код. 
2. Собери Maven-проект: подключи Jacoco, Mockito и JUnit.
3. Класс Lion не должен зависеть от класса Feline. Используй принцип инъекции зависимостей.
4. Напиши моки с помощью Mockito. Какие именно понадобятся — определи самостоятельно.
5. Напиши тесты на классы Feline, Cat и Lion.
6. Подумай, где можно применить параметризацию. Реализуй параметризованные тесты.
7. Оцени покрытие с помощью Jacoco: оно должно быть не менее 100%.

977605 committed:
1. add tests: AnimalTest, CatTest, FelineTest,FelineParameterizedTest, LionTest
2. Class Lion was modified
3. pom.xml was modified: add jacoco, junit, mockito

977605 committed:
1. deleted .idea/ and target/
2. gitignore was modified: add Default ignored files

977605 committed:
add jacoco report

egorvas commented on src/test/java/AnimalTest.java:
```diff
@@ -0,0 +1,48 @@
+    public void getFoodReturnsTrow() throws Exception {
+        Animal animal = new Animal();
+        animal.getFood("ERROR");
+        String expected = "Неизвестный вид животного, используйте значение Травоядное или Хищник";
```
**Нужно исправить:** твой тест до этого места не доходит и заканчивается на месте где происходит эксепшен. Текст ошибки фактически не проверяется. Есть другие способы проверять эксепшен в том числе с возможностью проверки текста ошибки, рекомендую использовать их.

NataliaN951 replied:
добавлены изменения: с помощью блока try-catch улавливается ошибка, далее проверяется сообщение об ошибке

egorvas commented on src/test/java/FelineTest.java:
```diff
@@ -0,0 +1,28 @@
+    @Test
+    public void getKittensWithoutArgumentsReturnKittens() {
+        Feline feline = new Feline();
+        feline.getKittens();
```
**Нужно исправить:** этот вызов бесполезен

NataliaN951 replied:
этот вызов удален

977605 committed:
1. deleted test getKittensWithoutArgumentsReturnKittens()
2. modified getFoodReturnsTrow(): used try-catch to catch and test message of exception
3. modified jacoco report

977605 committed:
add jacoco report

NataliaN951 commented on src/test/java/LionTest.java:
```diff
@@ -15,9 +15,9 @@ public class LionTest {
-        lion.getKittens();
-        Mockito.verify(feline).getKittens();
+        Assert.assertEquals(1, lion.getKittens());
     }
```
Добавлены изменения с учетом пожеланий на платформе: "....мне очень хочется увидеть хотя бы один тест где ты используешь моки (Mockito.when), а не просто verify...."
